graphql-java-support/src/test/java/com/apollographql/federation/graphqljava/FederationTest.java

```diff
@@ -37,7 +38,7 @@ void testEmpty() {
                 "\n" +
                 "type _Service {\n" +
                 "  sdl: String!\n" +
-                "}\n", SchemaUtils.printSchema(federated));
+                "}", SchemaUtils.removeDirectives(SchemaUtils.printSchema(federated)).trim());
```

Comment:
Do you understand what change in GraphQL-Java is necessitating this?  Is this definitely just a testing issue?  Note that part of federation is making the `_Service.sdl` field which uses the GraphQL-Java schema printer to print some SDL and return it to users (see SchemaTransformer.sdl()), so if this upgrade affects the output of that field, that might be an issue.

Reply:
I don't. I just started looking at upgrading to 14, and it's a big breaking release. The incompatibility with the federation lib is currently blocking to do any testing with federated services, so it's also hard to say what else might be broken. If we can do some sort of snapshot release that would give a much better idea what else needs to happen.

Reply:
The change appears to be on purpose.

The code change is here: [PR](https://github.com/graphql-java/graphql-java/pull/1609).
[graphql-java # 1587](https://github.com/graphql-java/graphql-java/issues/1587) describes the rationale.

> Standard directives are not printed when the includeDirectives flag is set. This seems like a design choice ...
> However this causes issues when we have to input the SDL to other GraphQL services that are not running graphql-java that don't default to having those declared. Can we add these to the SDL?

This should be a safe change. Note that the directives that are seen in the output are standard GraphQL directives that are part of the spec (include, skip and deprecated).